### PR TITLE
ARROW-5872: [C++][Gandiva] Support mod(double, double) function in Gandiva

### PR DESCRIPTION
**Pull request by pprudhvi**

### CODE REVIEW

File: cpp/src/gandiva/precompiled/types.h [hunk 1]
```diff
@@ -121,6 +121,7 @@ double months_between_timestamp_timestamp(uint64, uint64);
 int32 mem_compare(const char* left, int32 left_len, const char* right, int32 right_len);
 
 int32 mod_int64_int32(int64 left, int32 right);
```
Comment (fsaintjacques) [inline cpp/src/gandiva/precompiled/types.h]: I'm curious to know why this one does not have a context.

Comment (bkietz) [reply cpp/src/gandiva/precompiled/types.h]: From the other mod functions (implemented with the `MOD_OP` macro) it looks like expected behavior on divide-by-zero is to return the left operand and set no error: https://github.com/apache/arrow/pull/4820/files#diff-8772261adb65719b136969f075f5f678R54

Comment (pprudhvi) [reply cpp/src/gandiva/precompiled/types.h]: looks like a bug. created ARROW-5882.

**Review by praveenbingo — APPROVED** [review]
+1 Thanks @pprudhvi for doing this

**Pull request closed without merging**